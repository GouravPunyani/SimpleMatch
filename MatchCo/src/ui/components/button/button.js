// @flow
import type { ViewStyle, TextStyle } from 'react-native'
import React from 'react'
import { TouchableOpacity } from 'react-native'
import { Text } from '../..'
import { palette, color, spacing, fontFamily } from '../../theme'
import { translator } from '../../../services/translate/translator'
import glam from 'glamorous-native'
import { mergeStyles } from '../../../lib/merge-styles'
import { firebase } from 'firebase'

/**
 * The button properties.
 */
export type ButtonProps = {
  /** The text  */
  text?: string,
  /** Use this translation key to write the text. */
  tx?: string,
  /** Translates the given key into the appropriate language. */
  translate: (key: string) => string,
  /** Fires when the button is pressed. */
  onPress: Function,
  /** Styles applied to the wrapper. */
  style?: ViewStyle,
  /** Styles applied to the text. */
  textStyle?: TextStyle,
  /** Is interaction disabled? */
  disabled: Boolean
};

/**
 * It's a button.  Hooray!
 */
export type ButtonType = {
  /**
   * A button that blends into the background.
   */
  Clear: React.SFC<ButtonProps>,

  /**
   * A button with Facebook's brand color.
   */
  Facebook: React.SFC<ButtonProps>
};

/**
 * The outer-most component which forms the base of the button.
 */
const Wrapper: TouchableOpacity = glam(TouchableOpacity)(
  {
    minHeight: 44,
    minWidth: 44,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: color.buttonBackground,
    paddingHorizontal: spacing.medium,
    marginVertical: spacing.small,
    borderRadius: 2,
    shadowRadius: 1,
    shadowOffset: { width: 1, height: 1 },
    shadowColor: palette.black,
    shadowOpacity: 0.5
  },
  (props: ButtonProps) => mergeStyles(
    props.style,
    props.disabled && { backgroundColor: color.buttonBackgroundDisabled }
  )
)

/**
 * The text displayed on the button.
 */
const ButtonText: Text = glam(Text)(
  {
    color: color.buttonText,
    fontFamily: fontFamily.nexaBold,
    fontSize: 15
  },
  (props: ButtonProps) => mergeStyles(
    props.style,
    props.disabled && { color: color.buttonTextDisabled }
  )
)

/**
 * It's a button.  Hooray!
 */
export const Button: ButtonType = translator((props: ButtonProps): React.SFC<
  ButtonProps
> => {
  const { style, textStyle, onPress, disabled, tx, text, translate } = props

  return (
    <Wrapper style={style} onPress={onPress} disabled={disabled}>
      <ButtonText style={textStyle} disabled={disabled}>
        {tx ? translate(tx) : text}
      </ButtonText>
    </Wrapper>
  )
})

Button.Clear = (props: ButtonProps) => (
  <Button
    {...props}
    style={{ backgroundColor: color.backgroundColor }}
    textStyle={{ color: color.brand }}
  />
)

Button.Facebook = (props: ButtonProps) => (
  <Button
    {...props}
    style={{ backgroundColor: palette.facebook }}
    textStyle={{ color: palette.white }}
  />
)
