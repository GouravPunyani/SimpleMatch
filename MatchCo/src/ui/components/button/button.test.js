// @flow

import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { Button as Comp } from './button'

test('renders correctly', () => {
  const tree = renderer.create(<Comp />)
  expect(tree).toMatchSnapshot()
})
