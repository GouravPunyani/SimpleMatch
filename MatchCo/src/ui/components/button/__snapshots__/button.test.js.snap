// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "white",
      "borderRadius": 4,
      "borderWidth": 1,
      "justifyContent": "center",
      "margin": 4,
      "minHeight": 44,
      "minWidth": 44,
      "opacity": 1,
      "paddingHorizontal": 10,
      "paddingVertical": 4,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#333333",
          "fontFamily": "Nexa",
          "fontSize": 14,
        },
      ]
    }
  />
</View>
`;
